src/library/scala/Option.scala

```diff
@@ -280,6 +394,13 @@ sealed abstract class Option[+A] extends Product with Serializable {
    * nonempty '''and''' `pf` is defined for that value.
    * Returns $none otherwise.
    *
+   * This is equivalent to:
+   * {{{
+   * option match {
+   *   case Some(x) if pf.isDefinedAt(x) => Some(pf(x))
```

Comment:
is there a reason not to use `lift` like the actual code does? i.e. `case Some(x) => pf.lift(x)`

Reply:
I figured a boolean predicate function like `isDefinedAt` was as much as one would want to expose to a beginner about partial functions here, so showing `lift` seems like more cognitive load rather than less.